### PR TITLE
View, edit and insert listing pages

### PR DESCRIPTION
**Pull request by gabiseabra**
## View

screen shot 2019-01-24 at 15 55 16

## Edit

screen shot 2019-01-24 at 15 55 23

## Edit Images

screen shot 2019-01-24 at 15 58 21

## Insert

screen shot 2019-01-24 at 15 54 48

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -22,16 +22,15 @@ app.prepare().then(() => {
...
-    const route = routes[pathname]
-    if (route) {
-      return app.render(req, res, route.page, query)
-    }
-    return handle(req, res)
+  Object.keys(routes).map((path) => {
+    server.get(path, (req, res) => {
+      const parsedUrl = parse(req.url, true)
+      return app.render(req, res, routes[path].page, parsedUrl.query)
+    })
```
Comment (gabriel-laet) [inline server/server.js]: 👍

File: components/listings/EditAddress/AddressProperties.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
+import _ from 'lodash'
```
Comment (lucas-tulio) [inline components/listings/EditAddress/AddressProperties.js]: We could replace this and import only the functions we're using.

File: components/listings/Gallery/styles.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      border: '2px dashed ' + theme.colors.grey
+    }} .slick-arrow {
+    z-index: 100;
+    width: auto;
+    height: auto;
+    &::before {
+      display: none;
+    }
+    &.slick-prev {
+      left: 20px;
```
Comment (lucas-tulio) [inline components/listings/Gallery/styles.js]: Use `theme`?

File: components/shared/Modal/index.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.onKeyboardDismiss, false)
+    document.removeEventListener('click', this.onFocusOut, false)
+    document.removeEventListener('touched', this.onFocusOut, false)
+  }
+
+  onKeyboardDismiss = (e) => {
+    const {visible, onDismiss} = this.props
+    if (visible && onDismiss && e.keyCode === 27) onDismiss(e)
```
Comment (lucas-tulio) [inline components/shared/Modal/index.js]: Suggestion: move this `27` key code to a named constant, just so everybody knows which key this is.

File: components/shared/AddressAutoComplete/index.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
+import _ from 'lodash'
```
Comment (lucas-tulio) [inline components/shared/AddressAutoComplete/index.js]: Use specific imports.

File: components/shared/Modal/index.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
+import _ from 'lodash'
```
Comment (lucas-tulio) [inline components/shared/Modal/index.js]: Use specific imports.

File: pages/listings/edit.js [hunk 7]
```diff
@@ -0,0 +1,151 @@
+import _ from 'lodash'
```
Comment (lucas-tulio) [inline pages/listings/edit.js]: Use specific imports.

File: components/listings/EditProperties/index.js [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+import Col from '@emcasa/ui-dom/components/Col'
+import Row from '@emcasa/ui-dom/components/Row'
+import Input from '@emcasa/ui-dom/components/Input'
+import Dropdown from '@emcasa/ui-dom/components/Dropdown'
+import FieldContainer from '@emcasa/ui-dom/components/Field'
+import Fieldset from 'components/shared/Fieldset'
+import ScoreLevels from 'config/scores'
+import {ScoreIcon} from './styles'
+
+const MIN_PRICE = 250000
```
Comment (lucas-tulio) [inline components/listings/EditProperties/index.js]: This is not a major problem right now but we should find a solution for these kinds of settings. We have the same value hardocded somewhere in the front-end :/

File: components/listings/EditAddress/AddressProperties.js (hunk 2 above)
Comment (gabiseabra) [reply components/listings/EditAddress/AddressProperties.js]: We're using `babel-plugin-lodash` which cherry-pick required lodash modules in the final bundle, no need to optimize imports.

**Pull request merged**